ztp/resource-generator/entrypoints/generator

```diff
@@ -0,0 +1,425 @@
+    for filePath in "${sourceFileArr[@]}"; do
+        is_expected_CR $filePath "PolicyGenTemplate"
+        if [[ $? -eq 0 ]]; then
+            policyGenTemplates+="${filePath} "
+        fi
+    done
+
+    if [[ -z $policyGenTemplates ]]; then
+        echo "Error: No PolicyGenTemplates found"  >&2
+        exit 1   
+    fi
+
+    echo "Processing PolicyGenTemplates: $policyGenTemplates" 
+    echo "Generating configuration CRs into $destDir ..."
+    $policyGenPath/PolicyGenTemplate \
+        -sourcePath $policyGenPath/source-crs \
+        -wrapInPolicy=${WRAP_IN_POLICY} \
+        -outPath $destDir \
+        $policyGenTemplates
+    if [[ $? -ne 0 ]]; then
+        echo "Error: failed to generate configuration CRs"  >&2
+        exit 1
+    fi
+}
+
+# Verify the given arguments
+function verify_arguments {
+    if [[ $# -gt 2 ]] || [[ $# -lt 1 ]]; then
+        echo "Error: accepts 2 args, received $#" >&2
+        exit 1
+    fi
+
+    srcPath=$1
+    # given SRC_PATH is a relative path to MOUNT_DIR
+    if [[ ! $srcPath = /* ]]; then
+        srcPath="${MOUNT_DIR}/${srcPath}"
+    fi
+
+    destDir=""
+    if [[ $# -eq 1 ]]; then
+        # DEST_DIR is not given, set the default one
+        destDir="${MOUNT_DIR}/${outDir}"
+    elif [[ $# -eq 2 ]]; then
+        destDir=$2
+        if [[ ! $destDir = /* ]]; then
+            # given DEST_DIR is a relative path to MOUNT_DIR,
+            # set the obsolute path of the given DEST_DIR
+            destDir="${MOUNT_DIR}/${destDir}"
+        fi
+    fi
+
+    srcPath=$(realpath -m $srcPath)
+    destDir=$(realpath -m $destDir)
+
+    #is_dir_mounted $srcPath
+    #is_dir_mounted $destDir
+ 
+    if [[ -f $srcPath ]]; then
+        # SRC_PATH is a file and exists
+        sourceFileArr=($srcPath)
+    elif [[ -d $srcPath ]]; then
+        # SRC_PATH is a dir and exists
+        shopt -s nullglob
+        sourceFileArr=($srcPath/*.yaml)
+    elif [[ ! -f $srcPath ]] && [[ ! -d $srcPath ]]; then
+        echo "Error: $srcPath is not a file or a directory" >&2
+        exit 1
+    else
+        echo "Error: $srcPath is not found" >&2
+        exit 1
+    fi
+}
+
+# Check if a directory is mounted
+function is_dir_mounted {
+    local dir=$1
+
+    local baseDir="/$(echo $dir | cut -d '/' -f2)"
+    egrep -q $baseDir /proc/mounts
+    if [[ $? -ne 0 ]]; then
+        echo "Error: $dir is not a container volume mounted from the host or MOUNT_DIR is not set with desired path"  >&2
+        exit 1
+    fi
+}
+
+# Check if a yaml contains expected kind of CR
+function is_expected_CR {
+    local file=$1
+    local expected_cr=$2
+    local kind=
+
+    kind=$(yq eval -o=y '.kind' $file | head -1)
+    if [ $kind != $expected_cr ]; then
+        echo "${file} does not have ${expected_cr} CR. Skipping..."
+        return 1
+    fi
+    return 0
+}
+
+# Copy user defined additional resources in SiteConfig to /home/ztp/tmp
+# if they are relative paths to the parent directory of SiteConfig file
```

Comment:
Why do these files need to be copied? Does this mean that SiteConfig is not using the correct search path for finding the files? It would seem like it should always be able to find the files relative to the SiteConfig that it is processing.

Reply:
Based on our SiteConfig code, in theory, the directory of additional manifests or bios path is the relative path to the directory where the siteconfig is called.
With kustomize, it should be relative to the kustomization.yaml
For this usecase, we need to either go to the siteconfigs path that mounted from host and execute siteconfig there or copy the resources to another directory we want to execute from. I couldn't recall why I didn't choose to go to the siteconfigs directory at the first place... but looks like it's no disavantage and much easier than copying

Also may be we want to restrict the extra manifest path in the SiteConfig CR to be relative to the siteconfig yaml instead of the running path? @imiller0 @lack

Reply:
I think I like the no-copy idea, too.

As for restricting extra-manifest, that may be a nice thing to do in order to prevent mistakes, sure.